src/IssueViz/Models/AnalysisIssueLocationVisualization.cs

```diff
@@ -25,23 +25,21 @@ namespace SonarLint.VisualStudio.IssueVisualization.Models
     internal interface IAnalysisIssueLocationVisualization
     {
         int StepNumber { get; }
-
-        bool IsNavigable { get; }
-
         IAnalysisIssueLocation Location { get; }
+        bool IsNavigable { get; set; }
```

Comment:
FYI I'm just writing down ideas as a I look at the code at the moment. Don't treat them as review comments that need to be actioned.

I agree this property needs to be dynamically settable. The buffer tagger should play a part in updating it - if it can't translate a span, it should update the navigability.
Maybe it should be a calculated property: if we have span then it implies that the location is navigable. No span, not navigable.

Reply:
It's [already](https://github.com/SonarSource/sonarlint-visualstudio/blob/c8d0305aee7ea570d4f03284fd8310bb7389791f/src/IssueViz/Editor/LocationNavigator.cs#L64) somewhat calculated based on if there is a span or not. The navigability is currently:
1. Open the document
2. Get the document text view
2. Calculate span
3. [IOutliningManager.ExpandAll](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.outlining.ioutliningmanager.expandall?view=visualstudiosdk-2019)
4. [IEditorOperations.SelectAndMoveCaret](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.operations.ieditoroperations.selectandmovecaret?view=visualstudiosdk-2019#Microsoft_VisualStudio_Text_Operations_IEditorOperations_SelectAndMoveCaret_Microsoft_VisualStudio_Text_VirtualSnapshotPoint_Microsoft_VisualStudio_Text_VirtualSnapshotPoint_Microsoft_VisualStudio_Text_Editor_TextSelectionMode_System_Nullable_Microsoft_VisualStudio_Text_Editor_EnsureSpanVisibleOptions__)

So the span is currently only one of element of the equation. Although, perhaps we can assume that if we managed to get the text view AND calculate the span, then we will most likely succeed in the other operations as well.